config/jobs/kubernetes-sigs/cluster-api/cluster-api-prowjob-gen.yaml

```diff
@@ -0,0 +1,145 @@
+      # Please also make sure to refer a version where a kindest/node image exists for (see https://github.com/kubernetes-sigs/kind/releases/)
+      kubernetesVersionManagement: "v1.26.6@sha256:6e2d8b28a5b601defe327b98bd1c2d1930b49e5d8c512e1895099e4504007adb"
+      # This value determines the minimum Kubernetes
+      # supported version for Cluster API management cluster
+      # and can be found by referring to [Supported Kubernetes Version](https://cluster-api.sigs.k8s.io/reference/versions.html#supported-kubernetes-versions)
+      # docs (look for minimum supported k8s version for management cluster, i.e N-3).
+      # To check the latest available envtest in Kubebuilder for the minor version we determined above, please
+      # refer to https://github.com/kubernetes-sigs/kubebuilder/tree/tools-releases.
+      kubebuilderEnvtestKubernetesVersion: "1.26.1"
+      upgrades:
+      - from: "1.24"
+        to: "1.25"
+      - from: "1.25"
+        to: "1.26"
+      - from: "1.26"
+        to: "1.27"
+      - from: "1.27"
+        to: "1.28"
+      - from: "1.28"
+        to: "1.29"
+      - from: "1.29"
+        to: "1.30"
+    release-1.6:
+      kubekinsImage: "gcr.io/k8s-staging-test-infra/kubekins-e2e:v20240111-cf1d81388e-1.28"
+      interval: "4h"
+      kubernetesVersionManagement: "v1.25.11@sha256:227fa11ce74ea76a0474eeefb84cb75d8dad1b08638371ecf0e86259b35be0c8"
+      kubebuilderEnvtestKubernetesVersion: "1.25.0"
+      upgrades:
+      - from: "1.23"
+        to: "1.24"
+      - from: "1.24"
+        to: "1.25"
+      - from: "1.25"
+        to: "1.26"
+      - from: "1.26"
+        to: "1.27"
+      - from: "1.27"
+        to: "1.28"
+      - from: "1.28"
+        to: "1.29"
+    release-1.5:
+      kubekinsImage: "gcr.io/k8s-staging-test-infra/kubekins-e2e:v20240111-cf1d81388e-1.27"
+      interval: "4h"
+      kubernetesVersionManagement: "v1.24.15"
+      kubebuilderEnvtestKubernetesVersion: "1.24.2"
+      upgrades:
+      - from: "1.22"
+        to: "1.23"
+      - from: "1.23"
+        to: "1.24"
+      - from: "1.24"
+        to: "1.25"
+      - from: "1.25"
+        to: "1.26"
+      - from: "1.26"
+        to: "1.27"
+      - from: "1.27"
+        to: "1.28"
+    release-1.4:
+      kubekinsImage: "gcr.io/k8s-staging-test-infra/kubekins-e2e:v20240111-cf1d81388e-1.26"
+      interval: "4h"
+      kubernetesVersionManagement: "v1.23.17"
+      kubebuilderEnvtestKubernetesVersion: "1.23.5"
+      upgrades:
+      - from: "1.21"
+        to: "1.22"
+      - from: "1.22"
+        to: "1.23"
+      - from: "1.23"
+        to: "1.24"
+      - from: "1.24"
+        to: "1.25"
+      - from: "1.25"
+        to: "1.26"
+      - from: "1.26"
+        to: "1.27"
+
+  # The generator will run each template per branch and use a filename friendly
+  # branch variable to format the target filename.
+  templates:
+  - name: "cluster-api-periodics.yaml.tpl"
```

Comment:
(Please in a follow-up PR)

I thin I would change the output format.

Today the resulting grouping is a bit strange

<img width="363" alt="image" src="https://github.com/kubernetes/test-infra/assets/4662360/75102c13-f03b-4a0e-9307-93d91bc40a8f">

Basically first all periodics then grouped by branch then normal + upgrade
Then all presubmits

I think I would put the branch at the beginning cluster-api-{branch}-periodics / cluster-api-{branch}-periodics-upgrades cluster-api-{branch}-presubmits